### ISSUE TITLE
Specific subdomain supported?

### ISSUE THREAD

**Issue by FluffyDiscord** [blechschmidt/certbot-dns-local#1]
I need to add SSL certificate using DNS challenge only for one specific subdomain, eg. `my.domain.com`. I have added DNS NS record `_acme-challenge.my.domain.com.` with value `my.domain.com` and the `my.domain.com` A record has my server ip, but the certbot fails on timeout

```log
$: sudo certbot certonly -a certbot-dns-local:dns-local -d my.domain.com
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator certbot-dns-local:dns-local, Installer None
Obtaining a new certificate
Performing the following challenges:
dns-01 challenge for my.domain.com
Waiting 0 seconds for DNS changes to propagate
Waiting for verification...
Challenge failed for domain my.domain.com
dns-01 challenge for my.domain.com
Cleaning up challenges
Cleaning up challenges
Cleaning up challenges
Some challenges have failed.

IMPORTANT NOTES:
 - The following errors were reported by the server:

   Domain: my.domain.com
   Type:   dns
   Detail: DNS problem: query timed out looking up TXT for
   _acme-challenge.my.domain.com
```

**Comment by blechschmidt** [comment]
Sorry for the late reply. I wasn't a watcher of this repository for some reason, so I did not receive a mail when you created the issue. I just discovered your issue by accident today.

This tool does not differentiate between second-level or third-level domains, so it should work for subdomains. I uploaded the most recent version of the code to PyPI, which I used to test the tool again. I could not reproduce your issue with that version. To debug the issue, I would suggest you perform a `tcpdump -i any -n "udp port 53"` and check if you see the DNS challenge requests coming in from Let's Encrypt. Maybe `ufw` or some other firewall blocking DNS is in your way.

This is what my test configuration looks like:
![image](https://github.com/blechschmidt/certbot-dns-local/assets/18330291/4a3f7dc4-c596-4e32-999c-795e306e6e23)

And there is the certificate:
https://crt.sh/?id=11018118270
